sale_order_confirm_partial/tests/test_sale_order_confirm_partial.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import exceptions
+from odoo.tests import TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestSaleOrderConfirmPartial(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env["ir.config_parameter"].set_param(
+            "sale_order_confirm_partial.enabled", True
+        )
+        cls.sale_order = cls.env["sale.order"].create(
+            {
+                "name": "SO-TEST",
+                "partner_id": cls.env.ref("base.res_partner_1").id,
+                "pricelist_id": cls.env.ref("product.list0").id,
+                "order_line": [
```

Comment:
This way of line creation doesn't return a valid id for each created line. This may play a bad trick on you. Because you will reference those lines by their position. So if line position will be changed by some other module it will lead to an incorrect test setup.
You should do something like this instead:
```
cls.so_line_1 = cls.env["sale.order.line"].create(...)
cls.so_line_2 = cls.env["sale.order.line"].create(...)
```

So later you can reference both those lines directly.